controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaInterceptor.java

```diff
@@ -86,36 +82,22 @@
         return this.handler.authorize(resource, AUTH_CONTEXT_TOKEN.get());
     }
 
-    public static String retrieveDelegationToken(String tokenSigningKey) {
-        PravegaInterceptor interceptor = INTERCEPTOR_OBJECT.get();
-        if (interceptor != null) {
-            return interceptor.getDelegationToken();
-        } else {
-            Map<String, Object> claims = new HashMap<>();
-
-            claims.put("*", String.valueOf(READ_UPDATE));
-
-            return Jwts.builder()
-                       .setSubject("segmentstoreresource")
-                       .setAudience("segmentstore")
-                       .setClaims(claims)
-                       .signWith(SignatureAlgorithm.HS512, tokenSigningKey.getBytes())
-                       .compact();
-            }
-    }
-
-    public void setDelegationToken(String resource, AuthHandler.Permissions expectedLevel, String tokenSigningKey) {
-        if (AUTH_ENABLED) {
-            Map<String, Object> claims = new HashMap<>();
-
-            claims.put(resource, String.valueOf(expectedLevel));
-
-            delegationToken = Jwts.builder()
-                                  .setSubject("segmentstoreresource")
-                                  .setAudience("segmentstore")
-                                  .setClaims(claims)
-                                  .signWith(SignatureAlgorithm.HS512, tokenSigningKey.getBytes())
-                                  .compact();
-        }
+    /**
+     * Retrieves a master token for internal controller to segmentstore communication.
+     * @param tokenSigningKey Signing key for the JWT token.
+     * @return A new master token which has highest privileges.
+     * Throws an error if this is called from a grpc context.
```

Comment:
I don't see where the logic to throw in such a context is.

Reply:
Updated the comment. It was out of sync with the code.